cxx-checks/src/main/java/org/sonar/cxx/checks/FunctionComplexityCheck.java

```diff
@@ -46,7 +47,8 @@
   offset = "10min",
   effortToFixDescription = "per complexity point above the threshold")
 public class FunctionComplexityCheck extends SquidCheck<Grammar> {
-  private static final int DEFAULT_MAX = 10;
+
+  private static final int DEFAULT_MAX = 8;
```

Comment:
But everything bellow 10 imo should not raise any issues

On Tue, Jun 30, 2015, 21:43 Jorge Costa <jmecosta@gmail.com> wrote:

> That's a good idea, so you need multiple rules
>
> On Tue, Jun 30, 2015, 21:34 Günter Wirth <notifications@github.com> wrote:
>
>> In
>> cxx-checks/src/main/java/org/sonar/cxx/checks/FunctionComplexityCheck.java
>> <https://github.com/wenns/sonar-cxx/pull/541#discussion_r33607384>:
>>
>> > @@ -46,7 +47,8 @@
>> >    offset = "10min",
>> >    effortToFixDescription = "per complexity point above the threshold")
>> >  public class FunctionComplexityCheck extends SquidCheck<Grammar> {
>> > -  private static final int DEFAULT_MAX = 10;
>> > +
>> > +  private static final int DEFAULT_MAX = 8;
>>
>> My understanding is: 1-10 = Low risk program 11-20 = Moderate risk 21-50
>> = High risk
>> 50 = Most complex and highly unstable method
>> Especially in OO development inside of methods 8 is a good value. Maybe
>> we should allow to define the rule more than once? So someone can e.g. set
>> 8 as minor and 21 as critical?
>> … <#msg-f:1505429898446414984_msg-f:1505429302904145784_>
>>  Am 30.06.2015 um 19:12 schrieb Jorge Costa <notifications@github.com>:
>> In
>> cxx-checks/src/main/java/org/sonar/cxx/checks/FunctionComplexityCheck.java:
>> > @@ -46,7 +47,8 @@ > offset = "10min", > effortToFixDescription = "per
>> complexity point above the threshold") > public class
>> FunctionComplexityCheck extends SquidCheck<Grammar> { > - private static
>> final int DEFAULT_MAX = 10; > + > + private static final int DEFAULT_MAX =
>> 8; any good reason to change this to 8... we usually use 12 — Reply to this
>> email directly or view it on GitHub.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/wenns/sonar-cxx/pull/541/files#r33607384>.
>>
>